src/intersections/planes.jl

```diff
@@ -0,0 +1,45 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    intersecttype(segment, plane)
+
+Compute the intersection of a segment `s` and a plane `p`
+See https://en.wikipedia.org/wiki/Line-plane_intersection
+"""
+function intersecttype(s::Segment{3,T}, p::Plane{3,T}) where {T}
+    # Convert the positions of the segment vertices and the plane origin to coordinates
+    sᵥ = coordinates.(vertices(s))
+    pₒ = coordinates(origin(p))
+
+    # Get the normal of the plane
+    n = normal(p)
+    
+    # Calculate components
+    ln = (sᵥ[2] - sᵥ[1]) ⋅ n
+    pₒn = pₒ ⋅ n
+  
+    # If ln is zero, the segment is parallel to the plane
+    if isapprox(ln, zero(T))
+        # If the numerator is zero, the segment is coincident
+        if isapprox((pₒ - sᵥ[1]) ⋅ n, zero(T), atol=atol(T))
+            return ContainedSegmentPlane(s)
```

Comment:
We need to be very precise with these terms, coincident, contained, overlapping, etc to avoid proliferation of names in the intersection algorithms. If you can double check the exact concept that is being used here, then we can try to follow the naming convention everywhere in the function.